### PR TITLE
 #2620 : Converted LibraryViewCallback.java to kotlin

### LINKED ISSUE #2620: Convert LibraryViewCallback.java to Kotlin
#### What changes you have made:

Converted Java file to Kotlin

### PR DESCRIPTION
**Pull request by anamansari062**
Fixes #2620

Converted LibraryViewCallback.java to kotlin

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryViewCallback.kt [hunk 1]
```diff
@@ -15,30 +15,24 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.zim_manager.library_view;
 
-import java.util.LinkedList;
-import org.kiwix.kiwixmobile.core.base.BaseContract;
-import org.kiwix.kiwixmobile.core.entity.LibraryNetworkEntity;
-import org.kiwix.kiwixmobile.core.entity.LibraryNetworkEntity.Book;
+package org.kiwix.kiwixmobile.zimManager.libraryView
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryViewCallback.kt:19]: @anamansari062 Also you need to change the package name `zimManager`and `librayView`.
Screen Shot 2021-11-30 at 22 51 35

Comment (anamansari062) [reply app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryViewCallback.kt:19]: @gouri-panda sure i will do the changes

**Comment by anamansari062** [comment]
**If i am changing the name to : zim_manager and library_view , its not letting me commit** 

Running lint...

> Configure project :app
[directory '/Users/anamansari/kiwix-android/app/build', directory '/Users/anaman> Task :app:detekt FAILED
Property 'complexity>LongParameterList>threshold' is deprecated. Use 'functionThreshold' and 'constructorThreshold' instead.
naming - 5min debt
        PackageNaming - [LibraryViewCallback.kt] at /Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryViewCallback.kt:19:1

Overall debt: 5min


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:detekt'.
> Build failed with 1 weighted issues (threshold defined was 0).

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
11 actionable tasks: 1 executed, 10 up-to-date

> Configure project :app
[directory '/Users/anamansari/kiwix-android/app/build', directory '/Users/anamanBUILD SUCCESSFUL in 1s
10 actionable tasks: 10 up-to-date
Static analysis found violations and attempted to autofix, please commit these autoformat changes

    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------
    If the build failed with '.../.gradle/daemon/5.6.1/custom/scr/customexample/info.json No File Found' then you do not have JAVA_HOME set to JDK8
    Please make sure JAVA_HOME is set to JDK8
    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------

**Comment by gouri-panda** [comment]
@anamansari062 You should use Android studio's Refactor ->rename tools.   It looks like previous error.

**Comment by anamansari062** [comment]
@gouri-panda  previous error as in?

**Comment by gouri-panda** [comment]
The question you asked in the Slack channel.

**Comment by anamansari062** [comment]
@gouri-panda  yes but longParameterList is not there now and about package naming, you suggested to use camel case for packages, but the name are not in that format

**Comment by gouri-panda** [comment]
You only did half work. You should refractor the whole package names, in
this way you'll refractor the files that are underneath in the packages.

>

**Comment by anamansari062** [comment]
> You only did half work. You should refractor the whole package names, in this way you'll refractor the files that are underneath in the packages.
> […](#)

Sure will do

**Comment by anamansari062** [comment]
@gouri-panda I refactored the package names to "zimManager" and "libraryView" , so all the files under them changed (the package name). Its giving me error while commiting.

Running lint...

> Configure project :app
[directory '/Users/anamansari/kiwix-android/app/build', directory '/Users/anaman> Task :app:ktlintMainSourceSetCheck FAILED
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt:39:1: Exceeded max line length (100) (cannot be auto-corrected)IDLE
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt:73:1: Exceeded max line length (100) (cannot be auto-corrected)IDLE
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt:98:1: Exceeded max line length (100) (cannot be auto-corrected)IDLE
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:87:1: Exceeded max line length (100) (cannot be auto-corrected)
"plain" report written to /Users/anamansari/kiwix-android/app/build/reports/ktlint/ktlintMainSourceSetCheck.txt

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:ktlintMainSourceSetCheck'.
> Process 'command '/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
3 actionable tasks: 1 executed, 2 up-to-date

> Configure project :app
> Task :app:ktlintMainSourceSetFormat FAILEDapp/build', directory '/Users/anaman<=------------> 10% EXECUTING [1s]
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt:39:1: Exceeded max line length (100) (cannot be auto-corrected)
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt:73:1: Exceeded max line length (100) (cannot be auto-corrected)
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt:98:1: Exceeded max line length (100) (cannot be auto-corrected)
/Users/anamansari/kiwix-android/app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:87:1: Exceeded max line length (100) (cannot be auto-corrected)
"plain" report written to /Users/anamansari/kiwix-android/app/build/reports/ktlint/ktlintMainSourceSetFormat.txt

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:ktlintMainSourceSetFormat'.
> Process 'command '/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
3 actionable tasks: 1 executed, 2 up-to-date
Static analysis found violations and attempted to autofix, please commit these autoformat changes

    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------
    If the build failed with '.../.gradle/daemon/5.6.1/custom/scr/customexample/info.json No File Found' then you do not have JAVA_HOME set to JDK8
    Please make sure JAVA_HOME is set to JDK8
    ---------------------------IMPORTANT FOR KIWIX DEVELOPERS----------------------------------------------

**Comment by s-ayush2903** [comment]
@gouri-panda Why do we want to change the package name here? It is a simple refactor PR and we shouldn't be concerned about package names(because doing so is completely out of the context of linked issue and the PR, also there are no benefits of changing name). 
@anamansari062 that LongParameterList info by detekt is a warning, it shouldn't be causing any obstruction in letting you commit, also if you find yourself in trouble then can always see the generated report as mentioned in the logs:

>  "plain" report written to /Users/anamansari/kiwix-android/app/build/reports/ktlint/ktlintMainSourceSetFormat.txt

And when stuck with detekt, you can look "Rule sets" in https://detekt.github.io/detekt/, followed by searching for the error you're interested in

**Comment by s-ayush2903** [comment]
Ah apologies, wished to enter the keybinding for the comment button, seems I went a char off.

**Comment by anamansari062** [comment]
> @gouri-panda Why do we want to change the package name here? It is a simple refactor PR and we shouldn't be concerned about package names(because doing so is completely out of the context of linked issue and the PR, also there are no benefits of changing name). @anamansari062 that LongParameterList info by detekt is a warning, it shouldn't be causing any obstruction in letting you commit, also if you find yourself in trouble then can always see the generated report as mentioned in the logs:
> 
> > "plain" report written to /Users/anamansari/kiwix-android/app/build/reports/ktlint/ktlintMainSourceSetFormat.txt
> 
> And when stuck with detekt, you can look "Rule sets" in https://detekt.github.io/detekt/, followed by searching for the error you're interested in

LongParamenterList was resolved in this pr. I am having problems with package name only

**Comment by gouri-panda** [comment]
@s-ayush2903 I agree. but in order to refractor the parent package name format, we also refractor the children's package name, otherwise, we can't pass the lint check.

>LongParamenterList was resolved in this pr. I am having problems with package name only

@anamansari062 You only have to change two package names. Use Android studio's functionality . Click on the package name 
Refractor-> Rename. And then Android studio will do the rest.

**Comment by anamansari062** [comment]
> @s-ayush2903 I agree. but in order to refractor the parent package name format, we also refractor the children's package name, otherwise, we can't pass the lint check.
> 
> > LongParamenterList was resolved in this pr. I am having problems with package name only
> 
> @anamansari062 You only have to change two package names. Use Android studio's functionality . Click on the package name Refractor-> Rename. And then Android studio will do the rest.

@gouri-panda i have refactored the name using android studio, but its not letting me commit. All the child packages and files  did get changed accordingly.

**Comment by gouri-panda** [comment]
@anamansari062 What's the error?

**Comment by anamansari062** [comment]
@gouri-panda 
FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':app:ktlintMainSourceSetCheck'.
Process 'command '/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

**Comment by gouri-panda** [comment]
@anamansari062 I can't find the error. If you have difficulty with the error, override the precommit lint check by --no-verify in commit message. Like this git commit -m "this is a commit message" --no-verify  .Then I can look over here.

**Comment by anamansari062** [comment]
@gouri-panda done

**Comment by gouri-panda** [comment]
@anamansari062  Here all error are small error. 

Exceed max line means, you should start over from next line.

**Comment by anamansari062** [comment]
@gouri-panda  see i have only changed few lines related to java to kotlin. Rest others were done by refactor tool. Is it necessary to change package name now? We can create a new issue based on that. This issue deals with java to kotlin, that i have done. I can keep the package name same as before if you want.

**Comment by gouri-panda** [comment]
If you want change package name in other issue and then start with this issue, Go ahead!

**Comment by anamansari062** [comment]
> If you want change package name in other issue and then start with this issue, Go ahead!

Okayy

**Comment by anamansari062** [comment]
@gouri-panda I am having problems with my forked repository. I will close this pr, fork again and upload  a proper pr.

**Pull request closed without merging**